packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -351,3 +384,151 @@ type UnionDataLookup = {
   resolveType: 'unionData'
   typeMap: { [templateName: string]: string }
 }
+
+const _indexContent = async (tinaSchema: TinaSchema, database: Database) => {
```

Comment:
The indexing process just steps through each record and records a "namespaced" version of what the given record's value is, and points to an array of all the records which have that value.

For example, if I have 2 movies which both debuted in 1990, I'd the process would generate a single record that stores those references, effectively acting as a "lookup" record:
```json
"__attribute__movie#movie#released#1990": [
  "content/movies/top-gun.md",
  "content/movies/lethal-weapon.md"
]
```